maps/southern_cross/events/wildlife_encounter.dm

```diff
@@ -0,0 +1,94 @@
+// This event sends random sif mobs at someone in the wilderness, unannounced.
+
+/datum/event2/meta/wildlife_encounter
+	name = "random encounter"
+	departments = list(DEPARTMENT_EVERYONE)
+	chaos = 10
+	event_type = /datum/event2/event/wildlife_encounter
+
+/datum/event2/meta/wildlife_encounter/get_weight()
+	var/explorers = metric.count_people_with_job(/datum/job/explorer) + metric.count_people_with_job(/datum/job/sar)
+
+	if(!explorers)
+		return 0
+	return 20 + explorers * 50
+
+/datum/event2/event/wildlife_encounter
+	var/mob/living/victim = null
+
+/datum/event2/event/wildlife_encounter/set_up()
+	var/list/potential_victims = list()
+	for(var/mob/living/L in player_list)
+		//if(!(L.z in get_location_z_levels()))
+		//	log_debug("Not on the right z-level")
+		//	continue // Not on the right z-level.
+
+		if(L.stat)
+			continue // Don't want dead people.
+		if(istype(get_turf(L), /turf/simulated/floor/outdoors) && istype(get_area(L),/area/surface/outside/wilderness))
+			potential_victims += L
+			log_debug("Found victim")
+
+	if(potential_victims.len)
+		victim = pick(potential_victims)
+
+/datum/event2/event/wildlife_encounter/start()
```

Comment:
You need to null `victim` at the end of this to avoid GC failures.

Reply:
Nulled it, I don't know how to use a weak reference in DM.

Is it similar to a weak pointer in C++?
/weakref/proc/resolve() returns the result locate(ref), so I'd call resolve() on my weak reference when I need to access it?

If there's documentation on how to use these, I can't find it. ):

Reply:
Yep, basically all you need to know is:
````
var/weakref/foo = weakref(someatom)
````
````
var/atom/someatom = foo.resolve()
````
The main caveats are that you check victim is the expected type (and hasn't been qdeleted) since references can be reused.

Reply:
Also, you never delete weakrefs, you just null them and let them GC.